device/src/mac.rs

```diff
@@ -35,6 +36,16 @@ impl AdrAnsTrait for AdrAns {
 }
 
 impl Mac {
+    pub fn set_confirmed(&mut self) {
+        self.confirmed = true;
+    }
+
+    pub fn is_confirmed(&mut self) -> bool {
```

Comment:
it's unexpected for me to see changes to state in an `is_` method. In order to reduce risk of future confusion, I think it should be called something more complicated or split into two methods.